core/src/main/scala/cats/MonadTrans.scala

```diff
@@ -0,0 +1,17 @@
+package cats
+
+/**
+ * A type class which abstracts over the ability to lift an M[A] into a
+ * MonadTransformer
+ */
+trait MonadTrans[MT[_[_], _]] {
+
+  /**
+   * Lift a value of type M[A] into a monad transformer MT[M, A]
+   */
+  def liftT[M[_]: Monad, A](ma: M[A]): MT[M, A]
```

Comment:
One thing Edward Kmett noted in a talk he gave on [monad homomorphisms](https://youtu.be/YTaNkWjd-ac) is that `MonadTrans` provides no constraint that the resulting `MT[M, A]` has a `Monad` instance. I believe in his talk he fixes this by adding a function taking the `Monad[M]` constraint and providing a `Monad[MT[M, ?]]` constraint. In Scala it might look something like:

```scala
implicit def mtMonad[M[_]: Monad]: Monad[MT[M, A]]
```

Would we want something like that here?